Userland/Libraries/LibPthread/pthread.cpp

```diff
@@ -492,6 +510,11 @@ int pthread_setschedparam([[maybe_unused]] pthread_t thread, [[maybe_unused]] in
 // NOTE: libgcc expects this function to exist in libpthread, even if it is not implemented.
 int pthread_cancel(pthread_t)
 {
+    while (!cleanup_handlers.is_empty()) {
+        auto handler = cleanup_handlers.take_first();
+        handler.routine(handler.argument);
+    }
+
     TODO();
```

Comment:
Not really, as we haven't yet implemented actually canceling threads. I just added the handler cleanup code above the `TODO` already because I wanted to ensure that this isn't forgotten down the line, and because a `// FIXME: Once this is implemented, don't forget to run the cleanup handlers as well.` felt kind of weird.